py/server/deephaven/table_factory.py

```diff
@@ -272,6 +285,70 @@ def delete(self, table: Table) -> None:
         except Exception as e:
             raise DHError(e, "delete data in the InputTable failed.") from e
 
+    def add_async(self, table: Table, on_success: Callable[[], None] = None,
+                  on_error: Callable[[Exception], None] = None) -> None:
+        """Asynchronously writes rows from the provided table to this input table. If this is a keyed input table,
+        added rows with keys that match existing rows will replace those rows. This method returns immediately without
+        waiting for the operation to complete. If the operation succeeds, the optional on_success callback if provided
+        will be called. If the operation fails, the optional on_error callback if provided will be called. If on_error
+        is not provided, a default callback function will be called that simply prints out the received exception.
```

Comment:
Are there order processing guarantees?

Reply:
Yes for the requests made on the same thread, and it is probably safe to assume that it is not a typical use pattern to add to an InputTable from multiple threads, in which case,  it should fall on the user to sync the threads if some kind of ordering needs to be achieved.

Reply:
Document the guarantees if they exist.

Reply:
Jianfeng's statement around ordering is accurate. That said, it's currently implementation-defined, rather than something the Java interface guarantees. The implementations are thread-safe, and that is to be expected; concurrent usage from multiple threads, however, gives no guarantees about ordering (nor should it).